drivers/flash/flash_shell.c

```diff
@@ -149,20 +159,98 @@ static int cmd_write(const struct shell *shell, size_t argc, char *argv[])
 
 	shell_print(shell, "Write OK.");
 
-	if (flash_read(flash_dev, w_addr, check_array, op_size) < 0) {
-		shell_print(shell, "Verification read ERROR!");
-		return -EIO;
+	if (verify == 1) {
+		for (int i = 0; i < argc - argstart; i++) {
+			check_array[i] = ~buf_array[i];
```

Comment:
I wonder why they did a ~ in the check array before. Seems like wasted cpu cycles to me. 
If you initialize check_array to zero, and you read back zeroes, without error code and you wanted to write zeroes, thats fine right?